docs/plugin-eval.md

```diff
@@ -0,0 +1,221 @@
+# Plugins
+
+## Overview
+
+Creating and evaluating AI plugins can be a challenging task, as they may interact with the environment, collaborate with other plugins, and provide domain-specific knowledge to the AI model. Moreover, the AI model should be able to use the plugin effectively under various constraints. This plugin evaluation framework aims to simplify the process of creating, evaluating, and integrating AI plugins, ensuring a balance between realism and ease of evaluation.
+
+This framework is designed for local evaluation of AI plugins, and it's not intended for deploying Language Model plugins in production. The primary goal is to assist developers, researchers, and contributors in creating and evaluating custom plugins for their use cases, as well as understanding how the AI model responds to the installation of new plugins.
+
+## Getting Started
+
+To create a new plugin, follow these steps:
+
+1. Familiarize yourself with the example plugins found in the `openai` folder, such as `openai/bird.py`, `openai/clock.py`, and `openai/text_analyzer.py`. These examples will help you understand the structure and requirements for creating a new plugin.
+
+2. Follow the plugin creation process detailed below.
+
+3. Submit a pull request with your new plugin and evaluation example(s).
+
+## Preparing Samples
+
+Evaluations with plugins enabled take an additional arguement - `plugins` - which is a list of plugins to enable. The evaluation framework will automatically instantiate the plugins from the registry for each sampling.  Please note, we currently require chat-style inputs for plugins.
```

Comment:
> Evaluations with plugins enabled take an additional arguement - `plugins`

Not entirely clear to me what arguments you are referring to here. For example inside of `Match` I see plugins being instantiated inside of `eval_sample` from that sample, there is no argument at the Evaluation level.